torch/_inductor/lowering.py

```diff
@@ -413,38 +413,54 @@ def inner_fn(index):
     return inner
 
 
-def make_foreach_pointwise(aten_fn, pw_fn):
+def make_foreach_pointwise(pw_fn, allow_alpha=False):
     def inner(*inputs: List[List[TensorBox]], alpha=1):
-        def is_dynamic(t):
-            return any(x.free_symbols for x in t.data.get_size())
+        def is_dynamic(*args):
+            return any(isinstance(t, TensorBox) and any(x.free_symbols for x in t.data.get_size()) for t in args)
+
+        def has_type_promotion(*args):
+            if len(args) < 2:
+                return False
+            else:
+                dtype = args[0].data.get_dtype()
```

Comment:
can you also make sure that div is handled correctly - integer tensor divided by integer tensor gives floating point. Directed tests would be helpful

Reply:
Yeah I'm going to add tests, I was working on figuring out a general way to do it but I think I'll just have separate suites for each type of overload